src/hoi4_world/states/hoi4_states_converter.cpp

```diff
@@ -274,8 +288,62 @@ hoi4::States CreateStates(const std::map<int, vic3::State>& vic3_states,
       }
 
       const int64_t total_manpower = vic3_state_itr->second.GetPopulation();
+      const float factories = static_cast<float>(vic3_state_itr->second.GetEmployedPopulation()) / 100'000.0F;
       for (const auto& province_set: final_connected_province_sets)
       {
+         int civilian_factories = 0;
+         int military_factories = 0;
+         int dockyards = 0;
+
+         if (state_owner)
+         {
+            if (!accumulator.contains(*state_owner))
+            {
+               accumulator[*state_owner] = {0, 0, 0};
+            }
+            auto& country_factories = accumulator[*state_owner];
+            if (IsStateCoastal(province_set, {}))
+            {
+               //		20% chance of dockyard
+               //		57% chance of civilian factory
+               //		23% chance of military factory
+               country_factories.military += 0.23F * factories;
+               country_factories.civilian += 0.57F * factories;
+               country_factories.docks += 0.20F * factories;
+            }
+            else
+            {
+               //		 0% chance of dockyard
+               //		71% chance of civilian factory
+               //		29% chance of military factory
+               country_factories.military += 0.29F * factories;
+               country_factories.civilian += 0.71F * factories;
+            }
+
+            const float applied_factories =
+                std::max(0.0F, country_factories.military) + std::max(0.0F, country_factories.civilian) +
+                (IsStateCoastal(province_set, {}) ? std::max(0.0F, country_factories.docks) : 0.0F);
+
+            for (int i = 0; static_cast<float>(i) < std::clamp(factories, applied_factories, 12.0F); i++)
```

Comment:
Somewhat confused why clamp at 12 in particular. Vic2ToHoI4 hardcoded value? Feel there should be some sort of a config input for that.

Reply:
It's a Hoi4 limit. I used a constant, but I fear it's not more clear.